test/e2e/traces_test.go

```diff
@@ -0,0 +1,244 @@
+	tokenUrl := fmt.Sprintf("https://%s/dex/token", oidcEndpoint)
+	clientId := "test"                         // TODO Refactor this to a const shared with _helpers.go_
+	clientSecret := "ZXhhbXBsZS1hcHAtc2VjcmV0" // TODO Refactor this to a const shared with _helpers.go_
+
+	api, err := newObservatoriumAPIService(
+		e,
+		withGRPCListenEndpoint(":8317"),
+		withOtelTraceEndpoint(internalOtlpEndpoint),
+
+		// This test doesn't actually write logs, but we need
+		// this because Observatorium currently MUST see a logs or metrics endpoints
+		withLogsEndpoints("http://localhost:8080"),
+	)
+	testutil.Ok(t, err)
+	testutil.Ok(t, e2e.StartAndWaitReady(api))
+
+	// Warn if a YAML change introduced a tab character
+	if strings.ContainsRune(otelForwardingConfig, '\t') {
+		t.Fatalf("Tab in the YAML")
+	}
+
+	forwardingConfig := strings.Replace(otelForwardingConfig,
+		"{{OBS_CLIENT_ID}}",
+		clientId, -1)
+	forwardingConfig = strings.Replace(forwardingConfig,
+		"{{OBS_CLIENT_SECRET}}",
+		clientSecret, -1)
+	forwardingConfig = strings.Replace(forwardingConfig,
+		"{{OIDC_TOKEN_URL}}",
+		tokenUrl, -1)
+	forwardingConfig = strings.Replace(forwardingConfig,
+		"{{OBS_GRPC_ENDPOINT}}",
+		api.InternalEndpoint("grpc"), -1)
+	forwardingConfig = strings.Replace(forwardingConfig,
+		"{{DEX_TOKEN}}",
+		token, -1)
+
+	otelFile, err := ioutil.TempFile(e.SharedDir(), "fwd-coll.yaml")
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+
+	defer os.Remove(otelFile.Name())
+
+	_, err = otelFile.Write([]byte(forwardingConfig))
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+
+	// otelFile.Name() will give relative pathname to temp file.  Docker will complain with
+	// "If you intended to pass a host directory, use absolute path.""
+	otelFileName, err := filepath.Abs(otelFile.Name())
+	if err != nil {
+		t.Fatalf("unexpected Abs() error: %s", err)
+	}
+
+	otel := e.Runnable("otel-fwd-coll").
+		WithPorts(
+			map[string]int{
+				"http": 4318,
+				"grpc": 4317,
+			}).
+		Init(e2e.StartOptions{
+			// Note that if the forwarding collector was OIDC flow we would need
+			// "otel/opentelemetry-collector-contrib:0.45.0" instead.
+			Image:   "otel/opentelemetry-collector:0.45.0",
+			Volumes: []string{fmt.Sprintf("%s:/conf/collector.yaml", otelFileName)},
+			Command: e2e.Command{
+				Args: []string{"--config=/conf/collector.yaml"},
+			},
+		})
+
+	testutil.Ok(t, e2e.StartAndWaitReady(otel))
+
+	// Send trace insecurly to forwarding OTel collector for forwarding through Observatorium
+	// (This code could be refactored to observatorium/up, the test client)
+
+	client := &http.Client{}
+	request, err := http.NewRequest(
+		"POST",
+		fmt.Sprintf("http://%s/v1/traces", otel.Endpoint("http")),
+		bytes.NewBuffer([]byte(traceJSON)))
+	testutil.Ok(t, err)
+	request.Header.Set("Content-Type", "application/json")
+	response, err := client.Do(request)
+	testutil.Ok(t, err)
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
```

Comment:
Might be a good idea to replace this by `testutil.Equals(t, http.StatusOK, res.StatusCode)`.